### PR TITLE
feature/rpc: add rpc.type and rpc.status for metrics.

### PR DESCRIPTION
**Pull request by 1046102779**
https://github.com/open-telemetry/opentelemetry-specification/issues/2419

### CODE REVIEW

**Comment by trask** [comment]
hi @1046102779! I liked @arminru's https://github.com/open-telemetry/opentelemetry-specification/issues/2419#issuecomment-1074961559 proposing these as general metric attributes that can be used across all semantic conventions

File: semantic_conventions/trace/rpc.yaml [hunk 1]
```diff
@@ -4,6 +4,28 @@ groups:
     brief: 'This document defines semantic conventions for remote procedure calls.'
     events: [rpc.message]
     attributes:
+      - id: type
```
Comment (reyang) [inline semantic_conventions/trace/rpc.yaml:7]: For traces, the [span kind](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind) already has this information.

File: semantic_conventions/trace/rpc.yaml [hunk 2]
```diff
@@ -4,6 +4,28 @@ groups:
...
+        requires: always
+        brief: 'An RPC request consists of a caller and a callee, the name of the type is server or client.'
+        type:
+          allow_custom_values: false
+          members:
+            - id: server
+              value: 'server'
+            - id: client
+              value: "client"
+      - id: status
```
Comment (reyang) [inline semantic_conventions/trace/rpc.yaml:17]: Same here, the [span status](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#set-status) has covered this.

**Review by reyang — CHANGES_REQUESTED** [review]
I understand that the intention is to have something similar like [http.status_code](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/http-metrics.md#attributes) and such intention makes total sense to me. However, I thinking adding this to `semantic_conventions/trace/rpc.yaml` is going to introduce problem to traces as it introduces to span kind/status?

**Pull request closed without merging**